src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -331,6 +331,7 @@ public function beforeRender(Event $event)
 
         if ($this->ext && $isRecognized) {
             $this->renderAs($controller, $this->ext);
+            $response = $controller->response;
```

Comment:
this can be remove in future accidentally
adding a bit note is useful here IMO

Reply:
@saeideng The new test added should avoid such a regression. :slightly_smiling_face: